application/data/migrations/20200831000000_MutlipleDataTypes.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace Omeka\Db\Migrations;
+
+use Doctrine\DBAL\Connection;
+use Omeka\Db\Migration\MigrationInterface;
+
+class MutlipleDataTypes implements MigrationInterface
+{
+    public function up(Connection $conn)
+    {
+        $sql = <<<SQL
+ALTER TABLE `resource_template_property` CHANGE `data_type` `data_type` LONGTEXT COLLATE utf8mb4_unicode_ci DEFAULT NULL COMMENT '(DC2Type:json_array)';
+SQL;
+        $conn->exec($sql);
+
+        $sql = <<<SQL
+UPDATE `resource_template_property`
+SET `data_type` = NULL
+WHERE `data_type` IS NULL OR TRIM(`data_type`) = "";
+SQL;
+        $conn->exec($sql);
+
+        $sql = <<<SQL
+UPDATE `resource_template_property`
+SET `data_type` = CONCAT('["', REPLACE(TRIM(`data_type`), "\n", '","'), '"]')
```

Comment:
Quick question here: the handling of newlines and multiple values, that's accounting for a previous version of this that stored the multiple values that way?

I'd imagine we could leave that, and therefore the whole `REPLACE` out, yeah? And for that matter the check for `[` in the WHERE?

Anybody who's been on an intermediate version of one of these could just throw this migration id in the migrations table and skip it.

Reply:
Yes, it is related to an old version. I removed it.